File: backend/src/models/Bank.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IBank extends Document {
  userId: mongoose.Types.ObjectId;
  name: string;
  emailPatterns: string[];
  logo?: string;
  color: string;
  connected: boolean;
  lastActivity?: Date;
  createdAt: Date;
  updatedAt: Date;
}

const bankSchema = new Schema<IBank>(
  {
    userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    name: { type: String, required: true, trim: true },
    emailPatterns: [String],
    logo: String,
    color: { type: String, default: '#0D6B4F' },
    connected: { type: Boolean, default: true },
    lastActivity: Date,
  },
  { timestamps: true },
);

bankSchema.index({ userId: 1, name: 1 }, { unique: true });

export const Bank = mongoose.model<IBank>('Bank', bankSchema);

File: syncflow_app/lib/models/bank.dart
class Bank {
  final String id;
  final String name;
  final List<String> emailPatterns;
  final String? logo;
  final String color;
  final bool connected;
  final String? lastActivity;
  final String createdAt;

  Bank({
    required this.id,
    required this.name,
    this.emailPatterns = const [],
    this.logo,
    this.color = '#0D6B4F',
    this.connected = true,
    this.lastActivity,
    required this.createdAt,
  });

  factory Bank.fromJson(Map<String, dynamic> json) {
    return Bank(
      id: json['_id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      emailPatterns: (json['emailPatterns'] as List<dynamic>?)
              ?.map((e) => e as String)
              .toList() ??
          [],
      logo: json['logo'] as String?,
      color: json['color'] as String? ?? '#0D6B4F',
      connected: json['connected'] as bool? ?? true,
      lastActivity: json['lastActivity'] as String?,
      createdAt: json['createdAt'] as String? ?? '',
    );
  }
}
